symplyphysics/core/convert.py

```diff
@@ -1,19 +1,20 @@
 from sympy import Expr, sympify, S
-from sympy.physics.units import Quantity as SymQuantity
+# from sympy.physics.units import Quantity as SymQuantity
 
 from .dimensions import assert_equivalent_dimension
 from .symbols.quantities import Quantity
 
 
-def convert_to(value: SymQuantity, target_unit: SymQuantity) -> Expr:
+def convert_to(value: Expr, target_unit: Expr) -> Expr:
     """
     Convert ``value`` to its scale factor with ``value`` unit represented as ``target_unit``.
     """
-    target_quantity = Quantity(target_unit)
-    assert_equivalent_dimension(value, value.dimension.name, "convert_to",
+    value_quantity = value if isinstance(value, Quantity) else Quantity(value)
```

Comment:
When you use `isinstance`, its types should be present in function signature. Use `Expr | Quantity` for value type.

Reply:
But `Quantity | Expr` is virtually equivalent to `Expr` via MRO since `Quantity` is a subclass of `Expr`

Reply:
Agree.